### PR TITLE
N+1-integration-spec

### PR DESCRIPTION
**Pull request by gdumani**
- Install and configure Capybara
- Fix N+1 query
  - Before N+1 fix
beforen+1

  - After N+1 fix
after-n+1-comments

- Create integration tests for users
- Create integration tests for posts

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ def index
 
   def show
     @user = User.find(params[:user_id])
-    @post = @user.posts.find(params[:id])
+    @post = Post.find(params[:id])
     @comments = @post.comments
```
Comment (AnselemOdims) [inline app/controllers/posts_controller.rb]: - [ ] As stated in the requirements, you should try to solve the n+1 queries but your current implementation does not necessarily solve this. Please kindly ensure that you refactor to solve the `n+1` queries.

Hint:
`@posts = @user.posts.includes(:comments)`

File: spec/system/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      expect(page).to have_content('Comments: 1')
+    end
+
+    it 'shows how many likes a post has' do
+      expect(page).to have_content('Likes: 1')
+    end
+
+    it "redirects to a post's show page when clicking on it" do
+      click_link('>')
+      expect(page).to have_current_path(user_post_path(@user1.id, @post1.id))
```
Comment (AnselemOdims) [inline spec/system/posts_spec.rb]: - [ ] Good job writing these tests, but please kindly include test that asserts that you can see a section for pagination if there are more posts than fit on the view.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @gdumani , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (gdumani) [reply app/controllers/posts_controller.rb]: You are right, thank you very much!!

**Review by gemmen29 — APPROVED** [review]
Hi @gdumani, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**